M0228J-20 Add built-in objects content (#131)
MicroEJ/docs: ApplicationDeveloperGuide/microjs/builtin.rst

================

All standard built-in objects from ECMAScript 5.1 are available in MicroJS.
For the complete reference about these built-in objects, consult the `ECMA 5.1 specification <https://www.ecma-international.org/ecma-262/5.1/#sec-15>`_.

> For a description and example usage and of each method or property, consult a JavaScript documentation such as `Mozilla Developer Reference <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects>`_.

Reviewer: ```suggestion
For a description and usage examples of each method or property, consult a JavaScript documentation such as `Mozilla Developer Reference <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects>`_.
```

Excluded methods
----------------

> MicroJS supports every method and property in the ECMAScript built-in objects library (a total of **194** properties or methods) except **2** methods are excluded:

Reviewer: ```suggestion
MicroJS supports every methods and propertys in the ECMAScript built-in objects library (a total of **194** properties or methods) except **2** methods:
```

-  ``eval ( x )`` function
- ``Function ( p1, p2, … , pn, body )`` constructor

> The reason is that the JavaScript compiler is not embedded in MicroJS.

Reviewer: ```suggestion
The reason is that the JavaScript compiler is not embedded in the MicroJS runtime.
```

Built-in complete list
----------------------
- ✔️ = implemented
- 🕗 = in progress

> - ❌ = excluded

Reviewer: The documentation should document what is really done/available, so I think we should only list the implemented methods/properties. WDYT ?
